SCRD-5292 Added handling for encryptKey

guglo opened the pull request:
This commit includes the following features: 

1. Added a check if the cloud configuration is encrypted in Replace Server, Add Servers, Delete/Activate/Deactivate Compute,  Service Configuration and Service Info Page . 

2. If the cloud configuration is encrypted, when replace controller or compute server, the replace detail will have an extra required input for encrypt key; when add compute servers, the edit server detail will have an extra required input for encrypt key; when install OS, the select server page will have an extra required input for encrypt key; when update service configuration, it will have an extra required input for encrypt key.

3. If the cloud configuration is encrypted,  encryptKey has not been set, it will prompt user to input encrypt key. 

4.  If the cloud configuration is encrypted,  Replace Server, Add Servers , Delete/Activate/Deactivate Compute , isEncrypted and encryptKey are saved as global vars but not persisted. Also during playbook running process, if user refresh the process page, user will be prompted to re-enter encryptKey. 

5. If the cloud configuration is encrypted,  for Service Configuration and Service Info, during playbook running process, there is no mechanism to re-enter encryptKey when user refresh. 

6. Some minor lints, typo changes and missing method call fixes.

guglo commented:
Test encrypted can go with https://gerrit.suse.provo.cloud//#/c/5561
add in local.cfg to overwrite default false
[testing]
is_encrypted = true

eapodaca commented on src/localization/bundles/en.json:
```diff
@@ -69,6 +69,9 @@
     "get.passphrase.error": "Unable to use the provided SSH passphrase: {0}.\nPlease correct the problem and try again.",
     "id.with_colon": "ID:",
+    "common.encryptkey.message": "Please provide encryption key for running playbooks",
+    "common.encryptkey.title": "Provide Encryption Key",
```
Please use sentence case for this.

Commit:
rebased with latest master

SCRD-5292 Added handling for encryptKey

This commit includes:

1. Replace Server
  *isEncrypted and encrypteKey are saved as global vars but not peristed.
  *Replace Server details modal will have an input for encryptKey if
   clouddata isEncrypted.
  *When run playbook, it should have
   'extra-vars': {encrypt: this.props.encryptKey || ''}
   along with vault-password-file as part of parameters if isEncrypted.
  *During process, if user refresh the page, user will be prompted
   to re-enter encryptKey in a modal dialog if isEncrypted.

2. Add Server
  *isEncrypted and encrypteKey are saved as global vars but not peristed.
  *Edit Server details modal will have a required input for encryptKey if
   clouddata isEncrypted.
  *When run playbook, it should have
   'extra-vars': {encrypt: this.props.encryptKey || ''}
   along with vault-password-file as part of parameters if isEncrypted.
  *When click Deploy button without eidting a server, it will
   prompt user to enter encryptKey through a modal dialog if isEncrypted.
  *When click InstallOs, if isEncrypted, it will have a required
   input field for use to enter or modify encryptKey.
  *During process, if user refresh the page, user will be prompted
   to re-enter encryptKey in a modal dialog if isEncrypted.

3. Delete/Activate/Deactivate Compute
  *isEncrypted and encrypteKey are saved as global vars but not peristed.
  *When run playbook, it should have
   'extra-vars': {encrypt: this.props.encryptKey || ''}
   along with vault-password-file as part of parameters if isEncrypted.
  *When start, it will prompt user to enter encryptKey through
   a modal dialog if isEncrypted.
  *During process, if user refresh the page, user will be prompted
   to re-enter encryptKey in a modal dialog if isEncrypted.

4. Service Configuration
  *isEncrypted and encrypteKey are NOT saved as global vars.
  *If isEncrpted, a required input field will show up at right
   panel.
  *When run playbook, it should have
   'extra-vars': {encrypt: this.state.encryptKey || ''}
   along with vault-password-file as part of parameters if isEncrypted.
  *There is no refresh mechanism during playbooks running.

5. Service Info
  *isEncrypted and encrypteKey are NOT saved as global vars.
  *When user clicks run playbook to run playbook, it prompts user for
   encryptKey if isEncryted
  *When run playbook, it should have
   'extra-vars': {encrypt: this.state.encryptKey || ''}
   along with vault-password-file as part of parameters if isEncrypted.
  *There is no refresh mechanism during playbooks running.

6. Some minor lints, typo changes and missing method call fixes.

Commit:
Updated copyright year to be 2019

GarySmith commented on src/pages/UpdateServers.js:
```diff
@@ -386,6 +385,11 @@ class UpdateServers extends BaseUpdateWizardPage {
     // it to complete before moving on.
     await this.props.updateGlobalState('model', model);
 
+    // save the encryptKey in global cache so it could be retrieved later
```
I believe that the constructor needs something like:
```
  this.state = {
     ...this.state
```
in order to be consistent with the other components that extend the baseupdatewizard

guglo replied:
ok, will update.

GarySmith commented on src/UpdateWizard.js:
```diff
@@ -62,14 +63,18 @@ class UpdateWizard extends InstallWizard {
       // TODO how to deal with install password
       operationProps: undefined,
+
+      encryptKey: '',
```
It would seem that `isEncrypted` could be derived from `encryptKey`, e.g. it is exactly the same as the expression: `not isEmpty(encryptKey)`, and thus `isEncrypted` is basically duplicated data that should be removed. 

If I am misunderstanding this, then it is worth a comment in the code explaining why two variables are needed here instead of one.

guglo replied:
thanks @eapodaca  for the explanation. I will update with comments.

GarySmith commented:
Also, the commit comment could use some attention. It is hard to read, not line-wrapped correctly, has lots of repetition, and typos (like 'peristed' and 'encrpted').

guglo commented on src/pages/ReplaceServer/ReplaceController.js:
```diff
@@ -32,11 +32,12 @@ class ReplaceController extends BaseUpdateWizardPage {
       showLoadingMask: false,
 
-      ringBuilderConfirmation: undefined,
+      ringBuilderConfirmation: undefined
```
Not intentional..will restore it.

guglo commented:
@GarySmith  @eapodaca  thanks for the review. I will update the commit messages and address review comments.

Commit:
Merge branch 'master' into enkey

Commit:
Addressed review comments

Commit:
minor update

Commit:
Fixed minor styling issue

jeremy-moffitt commented:
tested without the key forced to on, will test next with it on... still need to piece through the code as well.... @guglo was there a real system with this enabled to test with too?

guglo commented:
@jeremy-moffitt  To test is_encrypted API call in development env, from ardana-service dir
mkdir data/ardana-ansible/group_vars
echo '$ANSIBLE_VAULT;1.1;AES256' >data/ardana-ansible/group_vars/all --> to test true
echo '1.1;AES256' > data/ardana-ansible/group_vars/all --> test false

jeremy-moffitt reviewed (approved):
tried it out locally, worked as advertised
